packages/commercetools/theme/pages/CheckoutV2/Payment.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <div>
+    <SfHeading
+      :level="3"
+      title="Billing"
+      class="sf-heading--left sf-heading--no-underline title"
+    />
+    <BillingForm
+      :handleBillingAddressSubmit="handleBillingAddressSubmit"
+      @stepSubmit="handleStepSubmit"
+    />
+  </div>
+</template>
+
+<script>
+import {
+  SfHeading
+} from '@storefront-ui/vue';
+import { useBilling } from '@vue-storefront/commercetools';
+import { onSSR } from '@vue-storefront/core';
+import BillingForm from '@/components/Checkout/BillingForm';
+
+export default {
+  name: 'Billing',
+  components: {
+    SfHeading,
+    BillingForm
+  },
+  setup(_, context) {
+    const {
+      save,
+      load
+    } = useBilling();
+
+    onSSR(async () => {
+      await load();
+    });
+
+    const handleBillingAddressSubmit = async billingDetails => {
```

Comment:
imho both this and `handlestepSubmit` are redundant abstractions here and we can just pass those directly in template (handleBillingAddressSubmit can be just renamed in return to ``saveBillingAddress`